src/OpenLoco/src/Map/TileElementBase.h

```diff
@@ -49,32 +52,42 @@ namespace OpenLoco::Map
             // Purposely clobers any other data in _type
             _type = enumValue(t) << 2;
         }
-        uint8_t flags() const { return _flags; }
+        ElementFlags flags() const { return _flags; }
         SmallZ baseZ() const { return _baseZ; }
         int16_t baseHeight() const { return _baseZ * kSmallZStep; }
         SmallZ clearZ() const { return _clearZ; }
         int16_t clearHeight() const { return _clearZ * kSmallZStep; }
 
-        bool isGhost() const { return _flags & ElementFlags::ghost; }
+        bool isGhost() const { return hasFlags(ElementFlags::ghost); }
         void setGhost(bool state)
         {
             _flags &= ~ElementFlags::ghost;
-            _flags |= state == true ? ElementFlags::ghost : 0;
+            _flags |= state == true ? ElementFlags::ghost : ElementFlags::none;
         }
-        bool isFlag5() const { return _flags & ElementFlags::flag_5; }
-        bool isFlag6() const { return _flags & ElementFlags::flag_6; } // in tracks/roads indicates is last tile of multi tile
+        bool isFlag5() const { return hasFlags(ElementFlags::flag_5); }
+        bool isFlag6() const { return hasFlags(ElementFlags::flag_6); } // in tracks/roads indicates is last tile of multi tile
         void setFlag6(bool state)
         {
             _flags &= ~ElementFlags::flag_6;
-            _flags |= state == true ? ElementFlags::flag_6 : 0;
+            _flags |= state == true ? ElementFlags::flag_6 : ElementFlags::none;
         }
         void setBaseZ(uint8_t baseZ) { _baseZ = baseZ; }
         void setClearZ(uint8_t value) { _clearZ = value; }
         bool isLast() const;
         void setLastFlag(bool state)
         {
             _flags &= ~ElementFlags::last;
-            _flags |= state == true ? ElementFlags::last : 0;
+            _flags |= state == true ? ElementFlags::last : ElementFlags::none;
+        }
+
+        constexpr bool hasFlags(ElementFlags flagsToTest) const
```

Comment:
not a task for this PR, but in future we should probably template this and `clearFlags` out, there are 12 existing enums all with this same code

Reply:
I agree with this, I actually proposed something like FlagsHolder<T> which adds all those things but duncan was a bit hesitant as it adds more template bloat, but I think its a reasonable solution.

Reply:
personally i'd prefer template bloat over macro bloat, but if he's against templates then is it possible to add these functions to the new enum macro?

Reply:
I'm not 100% against it. If we think its the best way forward we can do that. I don't think it will be too hard to change to that though so we can most likely continue as is for now whilst someone proposes a template idea.